test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/JsonPatchTests.cs

```diff
@@ -301,5 +301,22 @@ public async Task GivenAnExistingResourceWithMismatchingData_WhenPatchingWithPat
 
             Assert.Equal(HttpStatusCode.BadRequest, exception.Response.StatusCode);
         }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAPatchDocumentWithDateTime_WhenDateTimeHasOffset_ThenOffsetShouldBePreserved()
+        {
+            var poco = Samples.GetJsonSample("PatientWithMinimalData").ToPoco<Patient>();
+            FhirResponse<Patient> response = await _client.CreateAsync(poco);
+
+            string dateTimeOffsetString = "2022-05-02T14:00:00+02:00";
+            string patchDocument =
+                $@"[{{""op"":""add"",""path"":""/deceasedDateTime"",""value"":""{dateTimeOffsetString}""}}]";
+
+            FhirResponse<Patient> patchResponse = await _client.JsonPatchAsync(response.Resource, patchDocument);
+            Patient p = patchResponse.Resource;
+
+            Assert.Equal(dateTimeOffsetString, p.Deceased.ToString());
```

Comment:
I'm slightly worried `ToString` could swallow some issues in the future. Can you also check that `p.Deceased` contains the correct hour and timezone information?

Reply:
Good idea, I've added more robust test cases explicitly checking those fields. Let me know if you were thinking of a different approach though.